src/Samples/SubModelSeq/Program.fs

```diff
@@ -4,6 +4,24 @@ open System
 open Elmish
 open Elmish.WPF
 
+
+let mapIf p f a =
+  if p a then f a else a
+
+
+module Func =
+
+  let flip<'a, 'b, 'c> (f: 'a -> 'b -> 'c) =
+    fun b a -> f a b
+
+  let dedup (f: 'a -> 'a -> 'b) =
+    fun a -> f a a
```

Comment:
I find this confusing, and it's only used once. I suggest removing and rewriting relevant code.

Reply:
Like your suggestion for `Func.flip`, I have simplified the syntax to `let dedup f a = f a a`.

Unlike `Func.flip`, I am not in principle against inlining this function.  However, I find the one call site no clearer after inlining this, and I think this function and its use would be easier to understand if it had a better name.  I just asked [here](https://twitter.com/bender2k14/status/1286112662003363841) if anyone is aware of such a name.  I also admit that the call site might have other ways to improve clarity.

Reply:
>  I also admit that the call site might have other ways to improve clarity.

I'll leave this unresolved until we have investigated that. The call side mixes piping and composition, which I find confusing, so it may be better to rewrite it. We'll see.

Reply:
Actually, I'll just resolve this. If we modify the call side (to be reviewed separately) then we'll remove it anyway if it's no longer used.

Reply:
@jacobstanley was kind enough to point out [here](https://twitter.com/jacobstanley/status/1286593647639576576) that this function is `join` aka `flatten` for the monad `'a -> 'b` (in the type parameter `'b`).  I updated the name.